eo-maven-plugin/src/test/java/org/eolang/maven/UnplaceMojoTest.java

```diff
@@ -99,4 +104,59 @@ void testCleaning(@TempDir final Path temp) throws Exception {
             Matchers.is(false)
         );
     }
+
+    @Test
+    void testKeepBinaries(@TempDir final Path temp) throws Exception {
+        final Path foo = temp.resolve("a/b/c/foo5.class");
+        new Home().save("testKeepBinaries", foo);
+        final Path pparent = foo.getParent().getParent();
+        final Path list = temp.resolve("placed.csv");
+        Catalogs.INSTANCE.make(list)
+            .add(foo.toString())
+            .set(PlaceMojo.ATTR_KIND, UnplaceMojoTest.ATTR_KIND_CLASS)
+            .set(PlaceMojo.ATTR_RELATED, "a/b/c/foo5.class")
+            .set(PlaceMojo.ATTR_ORIGIN, "some-keep.jar")
+            .set(PlaceMojo.ATTR_HASH, new FileHash(foo));
+        new Moja<>(UnplaceMojo.class)
+            .with("placed", list.toFile())
+            .with("placedFormat", "csv")
+            .with("keepBinaries", new SetOf<>("**foo5.class"))
+            .execute();
+        MatcherAssert.assertThat(
+            new Home().exists(foo),
+            Matchers.is(true)
+        );
+        MatcherAssert.assertThat(
+            new Home().exists(Paths.get(String.valueOf(pparent))),
+            Matchers.is(true)
+        );
+    }
+
+    @Test
+    void testKeepRemoveBinaries(@TempDir final Path temp) throws Exception {
+        final Path foo = temp.resolve("a/b/c/foo6.class");
+        new Home().save("testKeepRemoveBinaries", foo);
+        final Path pparent = foo.getParent().getParent();
+        final Path list = temp.resolve("placed.csv");
+        Catalogs.INSTANCE.make(list)
+            .add(foo.toString())
+            .set(PlaceMojo.ATTR_KIND, UnplaceMojoTest.ATTR_KIND_CLASS)
+            .set(PlaceMojo.ATTR_RELATED, "a/b/c/foo6.class")
+            .set(PlaceMojo.ATTR_ORIGIN, "some-keep-remove.jar")
+            .set(PlaceMojo.ATTR_HASH, new FileHash(foo));
+        new Moja<>(UnplaceMojo.class)
+            .with("placed", list.toFile())
+            .with("placedFormat", "csv")
+            .with("keepBinaries", new SetOf<>("**foo6.class"))
+            .with("removeBinaries", new SetOf<>("**foo6.class"))
+            .execute();
+        MatcherAssert.assertThat(
+            new Home().exists(foo),
```

Comment:
@mximp this is against the logic explained in [the article](https://news.eolang.org/2022-07-15-placing-and-unplacing.html). First, we should delete everything except `foo6.class`. Then, we should delete `foo6.class`. In the end, the file should be gone.

Reply:
Didn't get the meaning of `removeBinaries`. Have changed the logic.